2022-01-10-report.org

```diff
@@ -0,0 +1,209 @@
+| TODO        |        24 |        33 |    +9 |
+| In Progress |         4 |         6 |    +2 |
+| Completed   |        90 |        94 |    +4 |
+
+*** Milestones Progress
+
+| Milestone                         | Last Week | This Week | Delta | Start Date     | Completion Date | Target         |
+|-----------------------------------+-----------+-----------+-------+----------------+-----------------+----------------|
+| Data Structures 1 - Core          |      100% |      100% | -     | 30th Nov 2020  | 27th Jan 2021   | 29th Jan 2021  |
+| Control Flow 1 - Core             |      100% |      100% | -     | 28th Jan 2021  | 10th Feb 2021   | 26th Feb 2021  |
+| Data Structures 2 - Generics      |      100% |      100% | -     | 11th Feb 2021  | 14th May 2021   | 28th May 2021  |
+| Data Structures 3 - Traits        |      100% |      100% | -     | 20th May 2021  | 17th Sept 2021  | 27th Aug 2021  |
+| Control Flow 2 - Pattern Matching |      100% |      100% | -     | 20th Sept 2021 | 9th Dec 2021    | 29th Nov 2021  |
+| Macros and cfg expansion          |        0% |        0% | -     | 1st Dec 2021   | -               | 28th Mar 2022  |
+| Imports and Visibility            |        0% |        0% | -     | 29th Mar 2022  | -               | 27th May 2022  |
+| Const Generics                    |        0% |        0% | -     | 30th May 2022  | -               | 25th Jul 2022  |
+| Intrinsics and builtins           |        0% |        0% | -     | 6th Sept 2021  | -               | 30th Sept 2022 |
+
+*** Risks
+
+| Risk                    | Impact (1-3) | Likelihood (0-10) | Risk (I * L) | Mitigation                                                 |
+|-------------------------+--------------+-------------------+--------------+------------------------------------------------------------|
+| Rust Language Changes   |            3 |                 7 |           21 | Keep up to date with the Rust language on a regular basis  |
+| Going over target dates |            3 |                 5 |           15 | Maintain status reports and issue tracking to stakeholders |
+
+** Planned Activities
+
+- Finish planning work for CFG and Macro expansion
+- Plan some GSoC 2022 projects
+- Complete bug fixes for Method Resolution
+- Begin work on Slices
+
+** Detailed changelog
+
+*** Add overflow checking on Literal's
+
+This checks that the literal value is within the bounds of their respective types. I have omitted code fixing the other issue in the bug report that overflow/max_val integers should be saturated to infinity when cast to REAL_TYPE's this seems like something we really should have documentation to reference in the code as to why this is the correct Rust behaviour.
+
+#+BEGIN_SRC rust
+fn test() -> i32 {
+    return 10000000000000000000000000000000000000000000;
+}
+#+END_SRC
+
+#+BEGIN_SRC
+<source>:2:12: error: decimal overflows the respective type 'f32'
+    2 |     return 10000000000000000000000000000000000000000000.0f32;
```

Comment:
Is this the right error?
```suggestion
<source>:2:12: error: decimal overflows the respective type 'i32'
    2 |     return 10000000000000000000000000000000000000000000;
```

Reply:
Yes you are right i copied and pasted the wrong one https://godbolt.org/z/oMWvoch56